addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -1232,8 +1253,11 @@ export class PosStore extends WithLazyGetterTrap {
             if (data["pos.session"].length > 0) {
                 // Replace the original session by the rescue one. And the rescue one will have
                 // a higher id than the original one since it's the last one created.
-                const session = this.models["pos.session"].sort((a, b) => a.id - b.id)[0];
-                session.delete();
+                const sessions = this.models["pos.session"].getAll().sort((a, b) => b.id - a.id);
```

Comment:
while you're checking the productsToDisplay, can you also check this? You should delete the old session so we should keep the previous ordering (`(a, b) => a.id - b.id`).

```suggestion
                const sessions = this.models["pos.session"].getAll().sort((a, b) => a.id - b.id);
```